### PR TITLE
allow to lock own artefacts as an editor

### PR DESCRIPTION
**Pull request by banglashi**
The feature requested was to allow users with the editor's role to lock/unlock their artefacts. A user with an admin role can still unlock them. I hope it makes sense to solve it this way and it has no other implications that I am unaware. Thanks.

### CODE REVIEW

File: public/javascripts/spacedeck_sections.js [hunk 1]
```diff
@@ -2524,11 +2525,18 @@ var SpacedeckSections = {
...
 
     show_toolbar_props: function() {
-      if (this.selection_metrics.count==0) return;
+      if (this.selection_metrics.count==0) {
+        this.toolbar_lock_in = false;
+        return;
+      }
       arts = this.selected_artifacts();
+      // check if selected artifacts are all from the same user
+      let userArts = true;
```
Comment (mntmn) [inline public/javascripts/spacedeck_sections.js]: nit: please rename to `same_user`, for example, to stay in line with other code style (even if idiosyncratic).

File: views/partials/tool/object-options.html [hunk 2]
```diff
@@ -7,12 +7,12 @@
 
     <button class="btn btn-divider"></button>
 
-    <button class="btn btn-transparent btn-icon-labeled" v-on:click="toggle_lock_of_selected_artifacts()" v-if="active_space_role=='admin'" title="<%=__("lock")%>">
+    <button class="btn btn-transparent btn-icon-labeled" v-on:click="toggle_lock_of_selected_artifacts()" v-if="active_space_role=='admin' || toolbar_lock_in">
```
Comment (mntmn) [inline views/partials/tool/object-options.html]: please manually merge latest `mnt` branch changes, there are now `lock_selected_artifacts` and `unlock_selected_artifacts` functions to make the buttons work predictably.

**Review by mntmn — CHANGES_REQUESTED** [review]

**Comment by banglashi** [comment]
fixed the 2 comments

**Comment by mntmn** [comment]
Thanks!

**Pull request merged**